src/controllers/colormapper.cpp

```diff
@@ -31,32 +31,47 @@ double colorDistance(QRgb a, QRgb b) {
 
 QRgb ColorMapper::getNearestColor(QRgb desiredColor) {
     // If desired color is already in cache, use cache entry
-    QMap<QRgb, QRgb>::const_iterator i = m_cache.constFind(desiredColor);
-    if (i != m_cache.constEnd()) {
-        DEBUG_ASSERT(m_availableColors.contains(i.value()));
-        kLogger.trace()
-                << "ColorMapper cache hit for" << desiredColor << ":"
-                << "Color =" << i.value();
-        return i.value();
+    const auto iCachedColor = m_cache.constFind(desiredColor);
+    if (iCachedColor != m_cache.constEnd()) {
+        const QRgb cachedColor = iCachedColor.key();
+        DEBUG_ASSERT(m_availableColors.contains(cachedColor));
```

Comment:
Just tested, i'm hitting this debug assertion.

Reply:
I think it should be iCachedColor.value(), not key()

Reply:
Tricky due to the same types. We should have test for this thing. But at least the debug assertion revealed it early.